src/app/forms/custom-sections/tyres/tyres.component.ts

```diff
@@ -41,24 +55,90 @@ export class TyresComponent implements OnInit, OnDestroy, OnChanges {
   get template() {
     switch (this.vehicleTechRecord.vehicleType) {
       case VehicleTypes.PSV:
-        return getTyresSection(VehicleTypes.PSV);
+        return tyresTemplatePsv;
       case VehicleTypes.HGV:
-        return getTyresSection(VehicleTypes.HGV);
+        return tyresTemplateHgv;
       case VehicleTypes.TRL:
-        return getTyresSection(VehicleTypes.TRL);
+        return tyresTemplateTrl;
     }
   }
 
+  get isPsv(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.PSV;
+  }
+
+  get isHgvOrTrl(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.HGV || this.vehicleTechRecord.vehicleType === VehicleTypes.TRL;
+  }
+
   get types(): typeof FormNodeEditTypes {
     return FormNodeEditTypes;
   }
 
+  get widths(): typeof FormNodeWidth {
+    return FormNodeWidth;
+  }
+
+  get speedCategorySymbol(): MultiOptions {
+    return speedCategorySymbolAsOptions();
+  }
+
+  get fitmentCode(): MultiOptions {
+    return fitmentCodeAsOptions();
+  }
+
   get axles(): CustomFormArray {
     return this.form.get(['axles']) as CustomFormArray;
   }
 
+  getAxleTyres(i: number): CustomFormGroup {
+    return this.axles.get([i, 'tyres']) as CustomFormGroup;
+  }
+
+  getTyresRefData(tyre: Tyres, axleNumber: number) {
+    this.isError = false;
+    this.referenceDataService
+      .fetchReferenceData(ReferenceDataResourceType.Tyres, String(tyre.tyreCode))
+      .pipe(take(1))
+      .subscribe({
+        next: data => {
+          const newTyre = {
```

Comment:
You can move this creation code of a Tyre interface into the interface model file, Ive seen this done in other typescript repos